Tests/Service/Request/RequestFactoryTest.php

```diff
@@ -160,12 +160,110 @@ public function testBuildFlow()
         );
     }
 
+    /**
+     * @return void
+     */
+    public function testBuildFlowWithQueryParams()
+    {
+        $baseUrl = 'baseUrl';
+        $routeString = '/routeString?param={param}';
+        $originParamValue = 'value with whitespaces';
+        $encodedParamValue = 'value+with+whitespaces';
+        $requestMethod = 'GET';
+        $requestBody = '{requestBody:requestBody}';
+        $endpointProphecy = $this->prophesize(EndpointInterface::class);
+        $endpointProphecy->__call('getBaseUrl', [])
+            ->willReturn($baseUrl)
+            ->shouldBeCalled()
+        ;
+        $endpointProphecy->__call('getPath', [])
+            ->willReturn($routeString)
+            ->shouldBeCalled()
+        ;
+        $endpointProphecy->__call('getMethod', [])
+            ->willReturn($requestMethod)
+            ->shouldBeCalled()
+        ;
+        $endpointProphecy->__call('getRequestFormat', [])
+            ->willReturn(EndpointInterface::FORMAT_JSON)
+            ->shouldBeCalled()
+        ;
+        $endpoint = $endpointProphecy->reveal();
+
+        $uri = $this->prophesize(UriInterface::class)->reveal();
+        $request = $this->prophesize(RequestInterface::class)->reveal();
+
+        $this->endpointRegistryProphecy
+            ->__call('getEndpoint', [Argument::type(ServiceRequestInterface::class)])
+            ->willReturn($endpoint)
+            ->shouldBeCalled()
+        ;
+
+        $this->serializerProphecy
+            ->__call('serialize', [Argument::type(ServiceRequestInterface::class), EndpointInterface::FORMAT_JSON])
+            ->willReturn($requestBody)
+            ->shouldBeCalled()
+        ;
+
+        $converterRoute = str_replace('{param}', $encodedParamValue, $routeString);
+        $this->uriFactoryProphecy
+            ->__call('createUri', [$baseUrl.$converterRoute])
+            ->willReturn($uri)
+            ->shouldBeCalled()
+        ;
+
+        $this->messageFactoryProphecy
+            ->__call('createRequest', [
```

Comment:
by the way, why not use 
`$this->messageFactoryProphecy->createRequest([...])` instead? I see it's already marked as `MessageFactory`, so IDE will autocomplete it and (IMO) it's more readable.

Reply:
I asked the same question when I opened the test the first time. And I absolutely agree with you about readability and IDE highlighting.
It's kinda style here, all tests are designed this way, and I decided to keep this style. Otherwise, I need to replace this construction everywhere (what I definitely want).